pennylane_lightning/core/src/simulators/lightning_tensor/cutn/MPSCutn.hpp

```diff
@@ -0,0 +1,341 @@
+
+        this->computeState(output_extentsPtr, output_tensorPtr);
+
+        std::vector<ComplexT> results(output_extent.front());
+        output_tensor.CopyGpuDataToHost(results.data(), results.size());
+
+        return results;
+    }
+
+  private:
+    /**
+     * @brief Return siteModes to the member initializer
+     *
+     * @return const std::vector<std::vector<std::size_t>>
+     */
+    const std::vector<std::vector<std::size_t>> setSitesModes_() {
+        std::vector<std::vector<std::size_t>> localSitesModes;
+        for (std::size_t i = 0; i < BaseType::getNumQubits(); i++) {
+            std::vector<std::size_t> localSiteModes;
+            if (i == 0) {
+                // Leftmost site (state mode, shared mode)
+                localSiteModes =
+                    std::vector<std::size_t>({i, i + BaseType::getNumQubits()});
+            } else if (i == BaseType::getNumQubits() - 1) {
+                // Rightmost site (shared mode, state mode)
+                localSiteModes = std::vector<std::size_t>(
+                    {i + BaseType::getNumQubits() - 1, i});
+            } else {
+                // Interior sites (state mode, state mode, shared mode)
+                localSiteModes =
+                    std::vector<std::size_t>({i + BaseType::getNumQubits() - 1,
+                                              i, i + BaseType::getNumQubits()});
+            }
+            localSitesModes.push_back(std::move(localSiteModes));
+        }
+        return localSitesModes;
+    }
+
+    /**
+     * @brief Return sitesExtents to the member initializer
+     *
+     * @return const std::vector<std::vector<std::size_t>>
+     */
+    const std::vector<std::vector<std::size_t>> setSitesExtents_() {
+        std::vector<std::vector<std::size_t>> localSitesExtents;
+
+        for (std::size_t i = 0; i < BaseType::getNumQubits(); i++) {
+            std::vector<std::size_t> localSiteExtents;
+            if (i == 0) {
+                // Leftmost site (state mode, shared mode)
+                localSiteExtents = std::vector<std::size_t>(
+                    {BaseType::getQubitDims()[i], maxBondDim_});
+            } else if (i == BaseType::getNumQubits() - 1) {
+                // Rightmost site (shared mode, state mode)
+                localSiteExtents = std::vector<std::size_t>(
+                    {maxBondDim_, BaseType::getQubitDims()[i]});
+            } else {
+                // Interior sites (state mode, state mode, shared mode)
+                localSiteExtents = std::vector<std::size_t>(
+                    {maxBondDim_, BaseType::getQubitDims()[i], maxBondDim_});
+            }
+            localSitesExtents.push_back(std::move(localSiteExtents));
+        }
+        return localSitesExtents;
+    }
+
+    /**
+     * @brief Return siteExtents_int64 to the member initializer
+     *
+     * @return const std::vector<std::vector<int64_t>>
+     */
+    const std::vector<std::vector<int64_t>> setSitesExtents_int64_() {
```

Comment:
```suggestion
    std::vector<std::vector<int64_t>> setSitesExtents_int64_() {
```

When returning a copy of data, we don't need to mark it as `const`, since it imposes a strictness on the outcome.

Reply:
Done!